Userland/Libraries/LibWeb/HTML/Origin.h

```diff
@@ -16,17 +16,20 @@ namespace Web::HTML {
 class Origin {
 public:
     Origin() = default;
-    Origin(DeprecatedString const& scheme, AK::URL::Host const& host, u16 port)
+    Origin(Optional<DeprecatedString> const& scheme, AK::URL::Host const& host, u16 port)
         : m_scheme(scheme)
         , m_host(host)
         , m_port(port)
     {
     }
 
     // https://html.spec.whatwg.org/multipage/origin.html#concept-origin-opaque
-    bool is_opaque() const { return m_scheme.is_null() && m_host.has<Empty>() && m_port == 0; }
+    bool is_opaque() const { return !m_scheme.has_value() && m_host.has<Empty>() && m_port == 0; }
 
-    DeprecatedString const& scheme() const { return m_scheme; }
+    StringView scheme() const
```

Comment:
Hmm, it looks like we don't care about nullability as far as I can tell by any caller, but I suppose either this should be nullable, or `m_scheme`'s should not be?

Reply:
StringView does have a null state.

Reply:
Right, but aren't we doing `value_or(""sv)` below?

Reply:
:thinking: I probably missed that, will check again.